src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -493,3 +542,27 @@ object IndexLogEntry {
 
   def schemaString(schema: StructType): String = schema.json
 }
+
+class FileIdTracker(private val startId: Long) {
+  private var maxId: Long = startId
+  private var fileNameToIdMap: mutable.HashMap[String, Long] = mutable.HashMap()
+
+  def getMaxFileId: Long = maxId
+
+  def getFileNameToIdMap: HashMap[String, Long] = fileNameToIdMap
+
+  def addSizeHint(size: Int): Unit = fileNameToIdMap.sizeHint(size)
```

Comment:
shouldn't this be `size` + the size of `fileNameToIdMap` unless `result` is already called? (not 100% sure)

Reply:
https://github.com/scala/scala/blob/9a369cbb1af27e6b01de54b16c2b054f9ac2b87f/src/library/scala/collection/mutable/HashMap.scala#L92 seems right?

Reply:
The code and documentation seems to say it is Ok. Let me know if you still suggest a change here.